src/NexusMods.Collections/InstallCollectionJob.cs

```diff
@@ -16,183 +14,220 @@
+        IServiceProvider provider,
+        LoadoutId target,
+        CollectionRevisionMetadata.ReadOnly revisionMetadata)
     {
-        // Collections are essentially zip files with a collection.json file in them along with several other files.
-        // So we can treat them as a library archive, and then extract the collection.json file from them.
-        if (!SourceCollection.AsLibraryFile().TryGetAsLibraryArchive(out var archive))
-            throw new InvalidOperationException("The source collection is not a library archive.");
-        
-        // Find the collection.json file and deserialize it.
-        var file = archive.Children.FirstOrDefault(f => f.Path == "collection.json");
-        CollectionRoot? root;
+        var connection = provider.GetRequiredService<IConnection>();
+        var datoms = connection.Db.Datoms(
+            (NexusModsCollectionLibraryFile.CollectionSlug, revisionMetadata.Collection.Slug),
+            (NexusModsCollectionLibraryFile.CollectionRevisionNumber, revisionMetadata.RevisionNumber)
+        );
+
+        if (datoms.Count == 0) throw new Exception($"Unable to find collection file for revision `{revisionMetadata.Collection.Slug}` (`{revisionMetadata.RevisionNumber}`)");
+        var source = NexusModsCollectionLibraryFile.Load(connection.Db, datoms[0]);
 
+        var monitor = provider.GetRequiredService<IJobMonitor>();
+        var job = new InstallCollectionJob
         {
-            await using var data = await FileStore.GetFileStream(file.AsLibraryFile().Hash);
-            root = await JsonSerializer.DeserializeAsync<CollectionRoot>(data, JsonSerializerOptions);
-        }
-        
-        if (root is null)
-            throw new InvalidOperationException("Failed to deserialize the collection.json file.");
+            TargetLoadout = target,
+            SourceCollection = source,
+            RevisionMetadata = revisionMetadata,
+            ServiceProvider = provider,
+            FileStore = provider.GetRequiredService<IFileStore>(),
+            Connection = connection,
+            LibraryService = provider.GetRequiredService<ILibraryService>(),
+            NexusModsLibrary = provider.GetRequiredService<NexusModsLibrary>(),
+        };
 
-        // The collection.json file includes the mods by various ids and other info, so first we'll link those to library archives.
-        ConcurrentBag<ModInstructions> toInstall = new();
+        return monitor.Begin<InstallCollectionJob, NexusCollectionLoadoutGroup.ReadOnly>(job);
+    }
 
-        await Parallel.ForEachAsync(root.Mods, context.CancellationToken, async (mod, _) => toInstall.Add(await EnsureDownloaded(mod)));
+    /// <summary>
+    /// Installs the collection.
+    /// </summary>
+    public async ValueTask<NexusCollectionLoadoutGroup.ReadOnly> StartAsync(IJobContext<InstallCollectionJob> context)
+    {
+        var isReady = CollectionDownloader.IsFullyDownloaded(RevisionMetadata, onlyRequired: true, db: Connection.Db);
```

Comment:
This will install the same collection multiple times, if it is already installed.
Is that the plan, or just the first iteration?

Is the expected behavior to instead verify whether it is already installed and perhaps install and missing mods in case it was only partially installed?

Reply:
Plan is to delete and reinstall collection in the future, since they are immutable.